plugins/frozen_column.cc

```diff
@@ -760,6 +762,349 @@ RegisterFrozenColumnFormatT<IntegerFrozenColumnFormat> regInteger;
 
 
 /*****************************************************************************/
+/* DOUBLE FROZEN COLUMN                                                     */
+/*****************************************************************************/
+
+/// Frozen column that stores each value as a signed 64 bit double
+struct DoubleFrozenColumn: public FrozenColumn {
+
+    struct SizingInfo {
+        SizingInfo(const TabularDatasetColumn & column)
+        {
+            if (!column.columnTypes.onlyDoublesAndNulls())
+                return;  // can't use this column type
+            numEntries = column.maxRowNumber - column.minRowNumber + 1;
+            hasNulls = column.sparseIndexes.size() < numEntries;
+
+            bytesRequired = sizeof(DoubleFrozenColumn) + numEntries * sizeof(Entry);
+        }
+
+        operator ssize_t () const
+        {
+            return bytesRequired;
+        }
+
+        ssize_t bytesRequired = -1;
+        size_t numEntries;
+        bool hasNulls;
+    };
+    
+    struct Entry {
+        
+        Entry()
+            : val(NULL_BITS)
+        {
+        }
+
+        Entry(double d)
+        {
+            U u { d: d };
+            val = u.bits;
+        }
+
+        uint64_t val;
+
+        static const uint64_t NULL_BITS
+            = 0ULL  << 63 // sign
+            | (0x7ffULL << 53) // exponent is all 1s for NaN
+            | (0xe1a1ULL); // mantissa
+
+        union U {
+            double d;
+            uint64_t bits;
+        };
+
+        bool isNull() const
+        {
+            return val == NULL_BITS;
+        }
+
+        double value() const
+        {
+            U u { bits: val };
+            return u.d;
+        }
+
+        operator CellValue() const
+        {
+            return isNull() ? CellValue() : value();
+        }
+    };
+
+    DoubleFrozenColumn(TabularDatasetColumn & column)
+        : columnTypes(column.columnTypes)
+    {
+        SizingInfo info(column);
+        ExcAssertNotEqual(info.bytesRequired, -1);
+
+        firstEntry = column.minRowNumber;
+        numEntries = info.numEntries;
+
+        // Check it's really feasible
+        ExcAssert(column.columnTypes.onlyDoublesAndNulls());
+        Entry * data = new Entry[info.numEntries];
+        storage = std::shared_ptr<Entry>(data, [] (Entry * p) { delete[] p; });
```

Comment:
Possibly dumb question, what happens if there is not enough memory to allocate the sared_ptr? Will we leak data? Then the whole program will crash so we don't care much?